pandas/core/algorithms.py

```diff
@@ -630,6 +634,9 @@ def factorize(
         uniques, codes = safe_sort(
             uniques, codes, na_sentinel=na_sentinel, assume_unique=True, verify=False
         )
+    if not dropna and (codes == na_sentinel).any():
```

Comment:
Can you assign `codes == na_sentinel` to a mask variable prior to this? Slightly preferable to re-evaluating this in the branch

Reply:
done, thanks!